crowbar_framework/app/models/service_object.rb

```diff
@@ -1069,6 +1069,8 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     proposal = Proposal.where(barclamp: @bc_name, name: inst).first
     save_proposal = false
 
+    nodes_previousbatch = nil
```

Comment:
Nitpick: please be consistent in your variable naming style.  "previous batch" is two words, not one.  Since the existing style is to put `_` between words, put it between all words ("nodes_previous_batch").  Also, since maximising legibility is an important goal, I prefer names which are as close as possible to natural English language, which is why I suggested `nodes_in_prev_batch` and `roles_for_prev_batch`.  Admittedly that's more of a personal preference, but still worth considering.